strr-platform-web/app/components/form/platform/BusinessDetails.vue

```diff
@@ -1,83 +1,43 @@
 <script setup lang="ts">
+import type { Form } from '#ui/types'
 const { t } = useI18n()
-const tPlat = (path: string) => t(`platform.${path}`)
-
-const { isComplete } = defineProps<{ isComplete: boolean }>()
-
 const { getBusinessSchema } = useStrrPlatformBusiness()
 const { platformBusiness } = storeToRefs(useStrrPlatformBusiness())
 
+const props = defineProps<{ isComplete: boolean }>()
+
+// cant set form schema type as businessDetailsSchema is a computed property
+const platformBusinessFormRef = ref<Form<any>>()
+
 const getRadioOptions = () => [
   { value: true, label: t('word.Yes') },
-  { value: false, label: t('word.No') }]
-
-// data manipulation stuff
+  { value: false, label: t('word.No') }
+]
 
+// reset cpbcLicenceNumber if hasCpbc radio button changed
 watch(() => platformBusiness.value.hasCpbc, () => {
   platformBusiness.value.cpbcLicenceNumber = ''
 })
 
-const businessdetailsSchema = computed(() => getBusinessSchema(
-  platformBusiness.value.hasCpbc, platformBusiness.value.hasRegOffAtt))
-
-watch(() => platformBusiness.value.regOfficeOrAtt.sameAsMailAddress, (val) => {
-  if (val) {
-    platformBusiness.value.regOfficeOrAtt.mailingAddress = { ...platformBusiness.value.mailingAddress }
-  }
-})
-watch(() => platformBusiness.value.mailingAddress, (val) => {
+// manage regOfficeOrAtt
+watchEffect(() => {
```

Comment:
This is a lot cleaner, but it looks like the inputs aren't updating as expected when toggling the checkbox.

Example: fill in mailing address, select 'yes registered office', trigger validation on reg office fields, check 'same as', uncheck 'same as', see validation is still err'd (its not updating properly anymore, which may be effecting the section validation and probably the form validation evaluation)